Dockerfile

```diff
@@ -0,0 +1,27 @@
+FROM python:3.9-slim-buster as base
+RUN apt-get update
+RUN apt-get install -y curl
+WORKDIR /app
+#RUN curl https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py
```

Comment:
Installing poetry in this fashion (as per the poetry docs) is slightly preferable - e.g. we'd lose our dependency on pip - but not a major issue. 
You may have found that you do need to add some environment variables though, so you might want something like..
```
ENV POETRY_HOME=/poetry
RUN curl https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
ENV PATH=/poetry/bin:${PATH}
```